tests-functional/tests/test_add_account.py

```diff
@@ -0,0 +1,148 @@
+import copy
+import pytest
+from resources.constants import new_account_data_1, new_account_data_2
+
+
+@pytest.mark.rpc
+class TestAddAccount:
+
+    @pytest.fixture(autouse=True)
+    def setup_backends(self, backend_new_profile):
+        self.account = backend_new_profile("sender")
+        self.account_data = copy.deepcopy(new_account_data_1)
+
+    def test_add_account_for_valid_key_uid(self):
+        self.account_data["key-uid"] = self.get_keypair_key_uid()
+        add_account_response = self.account.accounts_service.add_account(self.account_data)
+        self.account.verify_json_schema(add_account_response, method="accounts_addAccount")
+        assert "error" not in add_account_response
+
+        # After adding the account check that the get accounts will retrieve the new account
+        accounts_response = self.account.accounts_service.get_accounts()
+        accounts = accounts_response.get("result", [])
+        self.check_new_account_was_created(accounts, self.account_data)
+
+    def test_add_seconds_account_for_same_keypair(self):
+        key_uid = self.get_keypair_key_uid()
+
+        self.account_data["key-uid"] = key_uid
+        add_account_response = self.account.accounts_service.add_account(self.account_data)
+        assert "error" not in add_account_response
+
+        # Add a second account
+        self.account_data = new_account_data_2
+        self.account_data["key-uid"] = key_uid
+        add_second_account_response = self.account.accounts_service.add_account(self.account_data)
+        assert "error" not in add_second_account_response
+
+    def test_add_duplicate_account(self):
+        self.account_data["key-uid"] = self.get_keypair_key_uid()
+        add_account_response = self.account.accounts_service.add_account(self.account_data)
+        assert "error" not in add_account_response
+
+        # Add same account again
+        add_account_response = self.account.accounts_service.add_account(self.account_data, skip_validation=True)
+        assert add_account_response.get("error", {}).get("message", "") == "account already exists"
+
+    def test_add_new_account_via_add_keypair(self):
+        keypairs = self.get_account_keypairs()
+        keypair = keypairs[0]
+        self.account_data["key-uid"] = self.get_keypair_key_uid()
+
+        # Add the new account to the keypair accounts
+        keypair["accounts"] = [self.account_data]
+
+        # Call accounts_addKeypair with the new account
+        password = self.account.created_account.get("password")
```

Comment:
password is available with `self.account.password`

Then `created_account` is not needed. And I find it weird to save, as it's just a request to an endpoint.